src/export/bops/bops.test.ts

```diff
@@ -0,0 +1,187 @@
+    flow: {
+      _root: {
+        edges: ["user_role_question"],
+      },
+      user_role_question: {
+        type: 100,
+        data: {
+          fn: "user.role",
+          text: "user role",
+        },
+        edges: [
+          "applicant_id",
+          "agent_id",
+          "proxy_id",
+          "unsupported_id",
+          "other_id",
+        ],
+      },
+      applicant_id: {
+        type: 200,
+        data: {
+          text: "Applicant",
+          val: "applicant",
+        },
+      },
+      agent_id: {
+        type: 200,
+        data: {
+          text: "Agent",
+          val: "agent",
+        },
+      },
+      proxy_id: {
+        type: 200,
+        data: {
+          text: "Proxy",
+          val: "proxy",
+        },
+      },
+      unsupported_id: {
+        type: 200,
+        data: {
+          text: "Unsupported",
+          val: "unsupported",
+        },
+      },
+    },
+    sessionId: "123",
+    flowName: "Apply for a lawful development certificate"
+  };
+
+  [
+    ...USER_ROLES.map((role) => ({ passportValue: role, bopsValue: role })),
+    { passportValue: "unsupported", bopsValue: undefined },
+  ].forEach(({ passportValue, bopsValue }) => {
+    const expectedStr = JSON.stringify(bopsValue);
+
+    it(`sets { user_role: ${expectedStr} } when the passport user.role = '${passportValue}'`, () => {
+      const generatedPayload = getBOPSParams({
+        ...testParams,
+        breadcrumbs: {
+          user_role_question: {
+            auto: false,
+            answers: [`${passportValue}_id`],
+          },
+        },
+        passport: {
+          data: {
+            "user.role": [passportValue],
+          },
+        }
+      });
+      expect(generatedPayload.user_role).toEqual(bopsValue);
+    });
+  });
+});
+
+describe("works is set correctly based on the passport", () => {
+  const testParams = {
+    breadcrumbs: {},
+    flow: { _root: { edges: [] } },
+    sessionId: "123",
+    flowName: "Apply for a lawful development certificate"
+  };
+
+  it("sets start date in the payload when present in the passport", () => {
+    const generatedPayload = getBOPSParams({
+      ...testParams,
+      passport: {
+        data: {
+          "proposal.start.date": "2025-02-03",
+        },
+      },
+    });
+    expect(generatedPayload.works?.start_date).toEqual("2025-02-03");
+  });
+
+  it("sets completion date in the payload when present in the passport", () => {
+    const generatedPayload = getBOPSParams({
+      ...testParams,
+      passport: {
+        data: {
+          "proposal.completion.date": "2025-02-28",
+        },
+      },
+    });
+    expect(generatedPayload.works?.finish_date).toEqual("2025-02-28");
+  });
+
+  it("omits works from the payload when neither date is present in the passport", () => {
+    const generatedPayload = getBOPSParams({
+      ...testParams,
+      passport: {},
+    });
+    expect(generatedPayload).not.toHaveProperty("works");
+  });
+});
```

Comment:
I still need to port over all the various payload tests from the frontend here - especially around file-tagging and flags. 

Does one long test file feel like the right approach here (similar to how `export/oneApp` was ported over) or any objections to making a `exports/bops/tests/` subdirectory with files per payload property more like the current editor structure? I'm leaning towards the latter I think.